File: Python/a004.py
from sys import stdin

def isLeap(year):
    year = int(year)
    if year%400==0:
        return "閏年"
    elif year%100==0:
        return "平年"
    elif year%4==0:
        return "閏年"
    else:
        return "平年"

lst = [x for x in stdin]

print(*map(isLeap, lst), sep='\n')

File: CPP/a004.cpp
#include <iostream>
using namespace std;

int main(){
	int n;
	while (cin >> n){
		bool isLeap = false;
		if (n%400==0) isLeap=true;
		else if (n%100==0) isLeap=false;
		else if (n%4==0) isLeap=true;
		if (isLeap) cout << "閏年\n";
		else cout << "平年\n";
	}
	return 0;
}
